proposals/nnnn-atf-selenium-support.md

```diff
@@ -0,0 +1,236 @@
+
+#### Extend ExpectResponse
+
+Using the previous activate app example, the ExpectResponse method should be extended to be able to validate the state of the HMI changed post activating an app.
+
+### RPC Switching
+
+Because there is a mix of behavior for overwriting the previously mentioned methods in terms of requiring a simulated user action vs verifying the content displayed on the HMI, ATF should implement a structure that defines what actions the selenium web driver must take in the extended atf methods.
+
+Schema for RPC Switching feature
+
+The extended ExpectRequest method would reference this object to figure out what the selenium web driver actions are required. In this case the web driver would validate that the mainfield1 and mainfield2 text fields are displayed correctly
+
+```json
+{
+  "UI.Show" : {
+    "ExpectRequest": {
+      "validate": [{
+        "id": "#mainfield1", //user by Document.getElementById()
+        "text": <Show text 1> // text to validate the contents of the field
+      }, {
+        "id": "#mainfield2", //user by Document.getElementById()
+        "text": <Show text 2> // text to validate the contents of the field
+      }] 
+    }
+  }
+}
+```
+
+This example shows how an extended SendNotification for an onCommand would instead trigger a button press on the HMI causing the OnCommand to be sent.
+
+```json
+{
+  "UI.OnCommand" : {
+    "SendNotification": {
+      "action": [{
+        "id": "#cmdID20", //user by Document.getElementById()
+        "function": "click" // action to be taken by the webdriver
+      }] 
+    }
+  }
+}
+```
+
+This example shows how a SendRequests behavior would be overwritten for activating an app (which involves a user selecting the app from the app list)
+
+```json
+{
+  "SDL.ActivateApp" : {
+    "SendRequest": {
+      "action": [{
+        "id": "#appListItem1", //user by Document.getElementById()
+        "function": "click" // action to be taken by the webdriver
+      }] 
+    }
+  }
+}
+```
+
+
+This example shows how an ExpectResponse behavior would be overwritten for the activate app response. After the response is received the web driver will validate the app is active via the displayed app name
+
+```json
+{
+  "SDL.ActivateApp" : {
+    "ExpectResponse": {
+      "validate": [{
+        "id": "#appListHeader", //user by Document.getElementById()
+        "function": <App Name> // text to validate the contents of the field
+      }] 
+    }
+  }
+}
+```
+
+### Scope Of Test Scripts
+
+ATF has a very large collection of tests that are implemented with varying design patterns. It is hard to gauge the scope of how much work it would take to make every atf compatible with the selenium web driver. The author suggests that the initial implementation of this proposal covers basic smoke test cases such as registering an app, activating an app, displaying some data, and performing some sort of button press type interactions.
```

Comment:
```suggestion
ATF has a very large collection of tests that are implemented with varying design patterns. It is hard to gauge the scope of how much work it would take to make every ATF test compatible with the Selenium WebDriver. The author suggests that the initial implementation of this proposal covers basic smoke test cases such as registering an app, activating an app, displaying some data, and performing some sort of button press type interactions.
```